<launch>
    <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
    <arg name="node_namespace" default="apriltag" />

    <arg name="image_topic" default="image_raw" />
    <arg name="info_topic" default="camera_info" />

    <group ns="$(arg node_namespace)">
        <rosparam command="load" file="$(find bw_tags)/config/settings.yaml" ns="apriltag" />

        <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag" clear_params="true" required="false" output="screen" launch-prefix="$(arg launch_prefix)" >
            <remap from="camera_info" to="$(arg info_topic)" />
            <remap from="image_rect" to="$(arg image_topic)" />

            <param name="remove_duplicates" type="bool" value="true" />
            <param name="publish_tag_detections_image" type="bool" value="false" />
        </node>


        <node pkg="bw_tags" type="tag_marker_publisher.py" name="tag_marker_publisher" required="false" output="screen">
            <param name="base_frame" value="base_link"/>
        </node>
    </group>
</launch>
